src/Avalonia.Visuals/Animation/Transitions/Rotate3DTransition.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Avalonia.Collections;
+using Avalonia.Media;
+using Avalonia.Styling;
+
+namespace Avalonia.Animation;
+
+public class Rotate3DTransition: PageSlide
+{
+
+    /// <summary>
+    ///  Creates a new instance of the <see cref="Rotate3DTransition"/>
+    /// </summary>
+    /// <param name="duration">How long the rotation should take place</param>
+    /// <param name="orientation">The orientation of the rotation</param>
+    public Rotate3DTransition(TimeSpan duration, SlideAxis orientation = SlideAxis.Horizontal, double? depth = null)
+        : base(duration, orientation)
+    {
+        Depth = depth;
+    }
+    
+    /// <summary>
+    ///  Defines the depth of the 3D Effect. If null, depth will be calculated automatically from the width or height
+    ///  of the common parent of the visual being rotated.
+    /// </summary>
+    public double? Depth { get; set; }
+
+    /// <summary>
+    ///  Creates a new instance of the <see cref="Rotate3DTransition"/>
+    /// </summary>
+    public Rotate3DTransition() { }
+
+    /// <inheritdoc />
+    public override async Task Start(Visual? @from, Visual? to, bool forward, CancellationToken cancellationToken)
+    {
+        if (cancellationToken.IsCancellationRequested)
+        {
+            return;
+        }
+
+        var tasks = new List<Task>();
```

Comment:
As I see tasks can have a maximum of 2 elements. Can we use an array here instead of a list?

Reply:
Yes, we can. Copied this simply from the existing slide transition, which also uses a list.